### PR TITLE
Frontend conditional rendering

### PR DESCRIPTION
**Pull request by kiiamakir**
Added role-based authentication to the private route. This will take care of conditional rendering in the cases where some user roles shouldn't have access to a page. If multiple user groups are able to access a page but its content is should be modified based on the role of the user (for example in the front page) this should be done inside the component.

### CODE REVIEW

File: client/src/components/FrontPage.js [hunk 1]
```diff
@@ -10,33 +10,38 @@ import BasicGrid from './front-page/BasicGrid';
...
     coursesService.getCourses()
-      .then((data) => setPreviousCourses(data.courses.previous))
+      .then((data) => {
+        setCurrentCourses(data.courses.current);
+        setPreviousCourses(data.courses.previous);
+      })
       .catch((e) => console.log(e.message));
   }, []);
 
+  // Conditional rendering: only admin and teachers are shown the button for creating a new course
```
Comment (viivi-uhari) [inline client/src/components/FrontPage.js:31]: The commenting on this PR is very good! So, could this comment be of the same format as the others between components? As an example: CourseView.js, line 35

**Review by viivi-uhari — COMMENTED** [review]
Great job done with this PR! Didn't see anything major that would need changing.

**Review by marillam — APPROVED** [review]
Looks good to me!

**Comment by kiiamakir** [comment]
Conditional rendering still needs to be implemented for the new changes in main

**Review by viivi-uhari — APPROVED** [review]
Everything seems good and conditional rendering works!

**Pull request merged**